prow/jenkins/controller.go

```diff
@@ -254,12 +259,14 @@ func (c *Controller) terminateDupes(pjs []kube.ProwJob, jbs map[string]JenkinsBu
 			// Otherwise, abort it.
 			if buildExists {
 				if err := c.jc.Abort(toCancel.Spec.Job, &build); err != nil {
-					logrus.Warningf("Cannot cancel Jenkins build for prowjob %q: %v", toCancel.Metadata.Name, err)
+					c.log.WithFields(pjutil.ProwJobFields(&toCancel)).Warnf("Cannot cancel Jenkins build: %v", err)
 				}
 			}
 		}
 		toCancel.Status.CompletionTime = time.Now()
+		prevState := toCancel.Status.State
 		toCancel.Status.State = kube.AbortedState
+		c.log.WithFields(pjutil.ProwJobFields(&toCancel)).Infof("Transitioning from %q to %q", prevState, toCancel.Status.State)
```

Comment:
```go
WithFields(map[string]interface{}{"from": prevState, "to": toCancel.Status.State}).Info("Transitioning states.")
```

?? 

If we want to parse stuff like this in the tracer and have a nice summary of transitions or whatever might be easier if it's all in fields